textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -42,14 +88,160 @@ public void open() throws Exception {
 	 */
 	@Override
 	public ITuple getNextTuple() throws Exception {
-		// TODO Auto-generated method stub
+		if(innerTupleList.isEmpty()) {
+			return null;
+		}
+
 		ITuple nextTuple = null;
+
+		do {
+			if(getOuterOperatorNextTuple == true) {
+				if((outerTuple = outerOperator.getNextTuple()) != null) {
+					getOuterOperatorNextTuple = false;
+				} else {
+					return null;
+				}
+			}
+
+			if (innerOperatorCursor <= innerTupleList.size() - 1) {
+				innerTuple = innerTupleList.get(innerOperatorCursor);
+				innerOperatorCursor++;
+				if(innerOperatorCursor == innerTupleList.size()) {
+					innerOperatorCursor = 0;
+					getOuterOperatorNextTuple = true;
+				}
+			}
+
+			nextTuple = processTuples(outerTuple, innerTuple, joinPredicate);
```

Comment:
`processTuples()` was used in another function.  Can we rename it to `joinTuples()`?